File: Algorithms/1109/adjacent_difference.swift
class Solution {
    func corpFlightBookings(_ bookings: [[Int]], _ n: Int) -> [Int] {
        var opt = Array(repeating: 0, count: n)
        for it in bookings {
            opt[it[0] - 1] += it[2]
            if it[1] < n { opt[it[1]] -= it[2] }
        }
        for i in 1..<n {
            opt[i] += opt[i - 1]
        }
        return opt
    }
}

File: Algorithms/[1k, 2k)/1109/adjacent_difference.cpp
class Solution {
public:
    std::vector<int> corpFlightBookings(std::vector<std::vector<int>>& bookings, int n) {
        auto opt = std::vector<int>(n);
        for (; !bookings.empty(); bookings.pop_back()) {
            auto v = bookings.back();
            opt[v[0] - 1] += v[2];
            if (v[1] < n) opt[v[1]] -= v[2];
        }
        std::partial_sum(opt.begin(), opt.end(), opt.begin());
        return opt;
    }
};